assets/aml-benchmark/scripts/custom_inference_postprocessors/humaneval.py

```diff
@@ -170,13 +170,17 @@ def run_humaneval_postprocessor(
     # Post processing the prediction and ground truth columns
     for row in pred_dict_full:
         gt = "\n" + row["test"] + "\n" + "check(" + row["entry_point"] + ")"
-
         if str("def " + row["entry_point"] + "(") in row["original_prediction"]:
             # If the model regenerates the prompt/ function name
             pred_combined_prompt = row["original_prediction"]
         else:
-            pred_combined_prompt = row["prompt"]+"\n"+row["original_prediction"]
-
+            original_prediction = row["original_prediction"]
+            # If spaces were stripped from endpoint responses, add those back.
+            if not len(original_prediction) or (len(original_prediction) and original_prediction[0].isspace()):
```

Comment:
What if the prediction starts with 3/5 spaces instead of 4 spaces? Can we have a scenario like that?

Reply:
Here we check only the first character. If it is some whitespace character, that means endpoint is not stripping them. Else, we add the whitespace.

If there are 3-5 whitespaces, we do nothing, prefix is "".

Reply:
Yeah, I was wondering if in the future we get a scenario where the first sentence in the prediction is not indented correctly due to some reason which can cause the code to fail. The first sentence after the function declaration should generally start with "    ". So, in case it is not, we should check that and fix it. What do you think? As of now, we do not have this issue, so should be fine.